ts-defs-src/test-api/request.d.ts

```diff
@@ -0,0 +1,166 @@
+// Request API
+//----------------------------------------------------------------------------------------------------------------------
+
+type Method =
+    | 'get' | 'GET'
+    | 'delete' | 'DELETE'
+    | 'head' | 'HEAD'
+    | 'post' | 'POST'
+    | 'put' | 'PUT'
+    | 'patch' | 'PATCH';
+
+type Params = URLSearchParams | {[key: string]: string | number | boolean}
+
+interface AuthOptions {
+    username: string;
+    password: string;
+}
+
+interface ProxyOptions {
+    protocol?: string;
+    host: string;
+    port: number | string;
+    auth?: AuthOptions;
+}
+
+type RequestUrlOpts = string | URL | RequestOptions;
+
+interface RequestOptions {
+    /**
+     *  Request url.
+     */
+    url: string | URL;
+    /**
+     * Request methods.
+     */
+    method?: Method;
+    /**
+     * Request headers.
+     */
+    headers?: Record<string, string>;
+    /**
+     * Request params.
+     */
+    params?: Params;
+    /**
+     * Request body.
+     */
+    body?: object;
+    /**
+     * Request timeout.
+     */
+    timeout?: number;
+    /**
+     * Request credentials.
+     */
+    withCredentials?: boolean;
+    /**
+     * Request auth.
+     */
+    auth?: AuthOptions;
+    /**
+     * Request proxy.
+     */
+    proxy?: ProxyOptions;
+    /**
+     * Process response.
+     */
+    processResponse?: boolean;
```

Comment:
Need to discuss this option name. I propose the `rawResponse` name. Contact with @titerman .